src/ros_filter.cpp

```diff
@@ -1912,9 +1919,14 @@ namespace RobotLocalization
               " This was likely due to poorly coniditioned process, noise, or sensor covariances.");
       }
 
+      // If we're trying to publish with the same time stamp, it means that we had a measurement get inserted into the
+      // filter history, and our state estimate was updated after it was already published. As of ROS Noetic, TF2 will
+      // issue warnings whenever this occurs, so we make this behavior optional.
+      corrected_data = (!permitCorrectedPublication_ && lastPublishedStamp_ == filteredPosition.header.stamp);
```

Comment:
`lastPublishedStamp_ == filteredPosition.header.stamp` I don't think that works, because what if its older than just the last update. If you have something with substantial latency (100ms) but your filter is spinning at 100hz, then it could be much older than last, right?

Reply:
So I'm thinking aloud here. The filter history handles this situation:

- In the first cycle, the filter has four measurements to process. They have timestamps `1`, `2`, `4`, and `5`. They get processed, and we publish a state with timestamp `5`.
- In the second cycle, the filter receives only a measurement with timestamp `3`. It rewinds history to the posterior state after measurement `2` was fused, and then processes `3`, `4`, and `5` again. Then it publishes a state with timestamp `5`. This creates the issue in question, and is solved by this PR.

Now let's go over your scenario, or at least what I think you are saying.

- In the first cycle, the filter has four measurements to process. They have timestamps `1`, `2`, `4`, and `5`. They get processed, and we publish a state with timestamp `5`.
- In the second cycle, the filter has four measurements to process. They have timestamps `6`, `7`, `8`, and `9`. They get processed, and we publish a state with timestamp `9`.
- In the third cycle, the filter receives only a measurement with timestamp `3`. It rewinds history to the posterior state after measurement `2` was fused, and then processes `3`, `4`, `5`, `6`, `7`, `8`, and `9` again. Then it publishes a state with stamp `9` again. This also causes the issue, and is solved by this PR.

The filter was never publishing the updated state as it processed the history. It was rewinding, inserting, and processing all the way up to the latest measurement, then publishing. So I think checking the latest publication should work?

Note that in both of those cases, anyone listening to `tf` will get the exact same behavior before and after this change: the first transform published won't get updated when the second one arrives. It's just that in Noetic, we also get yelled at. :)

There is a third case.

- In the first cycle, the filter has four measurements to process. They have timestamps `1`, `2`, `4`, and `5`. They get processed, and we publish a state with timestamp `5`.
- In the second cycle, the filter has two measurements to process. They have timestamps `3` and `6`. The filter rewinds history to the posterior state after measurement `2` was fused, and then processes `3`, `4`, `5`, and `6`, and publishes the state with timestamp `6`. This is obviously not an issue.

Anyway, I will readily admit that I might be totally missing your point, and I apologize if so. Can you provide an example?

Reply:
OK, I wasn't sure where `filteredPosition`'s stamp was from, I didn't know that it would already be, at this point, rewinded back up to the latest message in all cases. That works. 

I might just make it a little more error proof with `lastPublishedStamp_ >= filteredPosition.header.stamp`. I agree it shouldn't matter, but I think it makes syntactically more clear why this is being checked (or why they might be equal).

Reply:
Yeah, that situation should never happen, but it also won't hurt to change it, and I agree with the increased clarity.